File: ios/UnflowReactNativeSdk.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(UnflowReactNativeSdk, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(initialize:(NSString)apiKey withEnableLogging:(BOOL)enableLogging)

RCT_EXTERN_METHOD(sync)

@end

File: ios/UnflowReactNativeSdk.swift
@objc(UnflowReactNativeSdk)
class UnflowReactNativeSdk: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }

    @objc(initialize:withEnableLogging:)
    func initialize(apiKey: String, enableLogging: Bool) -> Void {
        NSLog("UNFLOW - INITIALIZING")
    }

    @objc(sync)
    func sync() -> Void {
        NSLog("UNFLOW - SYNCING")
    }
}
